pkg/cri/resource-manager/visualizer/flags.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2019-2020 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package visualizer
+
+import (
+	"flag"
+	"strings"
+)
+
+// dirList is a comma-separated list of directory names to search for visualizers.
+type dirList string
+
+// Set registers external visualizer implementations found in the directory list.
+func (d *dirList) Set(value string) error {
+	for _, dir := range strings.Split(value, ",") {
+		log.Info("searching for external visualizers in %s...", dir)
+		visualizers.discoverExternal(dir)
```

Comment:
Just a comment: I think this is fine for now but I think it would be nicer to separate the discovery from flag parsing. But I guess that can be done later if we want to allow "hot-plugging" of visualizers, for example

Reply:
Hmm, you mean we'd try to look discover external visualizers on demand, without having to restart cri-resmgr ? That's a good idea. I'll update the code to do that.

Reply:
Yeah, that was my thinking. Cool, if you're implementing that 👍

Reply:
@marquiz I pushed a version implementing on-demand external UI discovery.

The current code does the discovery without caching any results. When an external UI is chosen with the normal browsing flow (-> index page -> select UI) we do discovery twice: once to generate the index page, then once when the external UI is selected. Caching the results should be easy to add if it is considered necessary.

Reply:
Cool, I think this is enough or now.